core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -88,7 +89,8 @@ public static InstallState getInstallState() {
             // Edge case: used Jenkins 1 but did not save the system config page,
             // the version is not persisted and returns 1.0, so try to check if
             // they actually did anything
-            if (!Jenkins.getInstance().getItemMap().isEmpty()) {
+            if (!Jenkins.getInstance().getItemMap().isEmpty()
+                    || Jenkins.getInstance().getSecurityRealm() != SecurityRealm.NO_AUTHENTICATION) {
```

Comment:
@daniel-beck it handles the normal 2.0 case but restarts may be an issue, I'll investigate this